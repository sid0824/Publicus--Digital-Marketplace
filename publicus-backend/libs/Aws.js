import AWS from 'aws-sdk';
import {AccessKeyId, SecretAccessKey} from "../config/ConfigValues";

export const s3 = new AWS.S3({
  region: 'ap-southeast-2',
  accessKeyId: AccessKeyId,
  secretAccessKey: SecretAccessKey
});

export const ses = new AWS.SES({
  region: 'us-west-2'
});