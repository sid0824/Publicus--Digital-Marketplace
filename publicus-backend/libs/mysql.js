import mysql from 'mysql';
import {MySqlHost, MySqlPassword, MySqlUser} from "../config/ConfigValues";

//mysql connection to database
const connection = mysql.createConnection({
    host: MySqlHost,
    user: MySqlUser,
    password: MySqlPassword,
    database: "survey",
    multipleStatements: true
});

export const runQuery = async (query) => {
    return new Promise((resolve, reject) => {
        connection.query(query, (error, results) => {
            if(error) {
                reject(error);
            } else {
                resolve(results);
            }
        });
    });
};