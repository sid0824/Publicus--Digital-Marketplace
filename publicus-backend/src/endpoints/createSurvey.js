import {runQuery} from "../../libs/mysql";
import {failure, success} from "../util/ResponseUtil";
import uuidv1 from 'uuid/v1';
import {insertTagsQuery} from "../util/QueryUtil";

const defaultImage = "https://designshack.net/wp-content/uploads/placeholder-image.png";

//Generate SQL query to insert Survey (adds new survey to database)
const insertSurveysQuery = (surveyId, userId, name, price, form, description, redirect, instructions) => {
  return `INSERT INTO Surveys (surveyId, userId, name, price, form, description, redirect, image, instructions) 
          VALUES( '${surveyId}', '${userId}', '${escape(name)}', ${price}, '${escape(form)}', '${escape(description)}', 
          '${escape(redirect)}', '${defaultImage}', '${escape(instructions)}' )`;
};

export async function main(event, context) {
  try {
    //Retrieve body variables
    const data = JSON.parse(event.body);
    const newSurveyId = uuidv1();

    // Add the survey to the database
    await runQuery(insertSurveysQuery(newSurveyId, event.requestContext.identity.cognitoIdentityId,
      data.name, data.price, JSON.stringify(data.form), data.description, data.redirect || '', data.instructions || ''));

    // Add the survey tags to the database
    if (data.tags && data.tags.length) {
      await runQuery(insertTagsQuery(newSurveyId, data.tags));
    }
    return success();
  } catch (err) {
    console.log(err);
    return failure();
  }

}
