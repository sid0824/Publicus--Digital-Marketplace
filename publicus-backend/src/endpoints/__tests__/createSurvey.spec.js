import { main } from '../createSurvey';
import { runQuery } from "../../../libs/mysql";
import uuidv1 from 'uuid/v1';

jest.mock("../../../libs/mysql");
jest.mock('uuid/v1');

const mockEvent = {
  requestContext: {
    identity: {
      cognitoIdentityId: 'MOCK_COGNITO_ID'
    },
  },
  body: JSON.stringify({
    name: 'NAME',
    price: 5,
    form: [{ MOCK: 'FORM' }],
    description: 'DESCRIPTION',
    redirect: 'REDIRECT',
    instructions: 'INSTRUCTIONS',
    tags: ["TAG1", "TAG2"]
  }),
  pathParameters: {
    surveyId: 'MOCK_SURVEY_ID'
  }
};

describe('createSurvey', () => {

  beforeEach(() => {
    runQuery.mockResolvedValue(null);
    uuidv1.mockReturnValue('575ea33e-f654-11e9-802a-5aa538984bd8');
  });

  afterEach(() => {
    runQuery.mockReset();
  });

  test(`should respond with failure when an error occurs`, async () => {
    runQuery.mockReset();
    runQuery.mockImplementation(() => {
      throw new Error('ERROR')
    });
    const response = await main(mockEvent);
    expect(response).toMatchSnapshot();
  });

  test(`should add the survey to the database`, async () => {
    const response = await main(mockEvent);
    expect(runQuery.mock.calls[0][0]).toMatchSnapshot();
    expect(runQuery.mock.calls[1][0]).toMatchSnapshot();
    expect(response).toMatchSnapshot();
  })

});