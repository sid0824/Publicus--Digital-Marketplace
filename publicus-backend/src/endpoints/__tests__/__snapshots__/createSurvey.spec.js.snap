// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`createSurvey should add the survey to the database 1`] = `
"INSERT INTO Surveys (surveyId, userId, name, price, form, description, redirect, image, instructions) 
          VALUES( '575ea33e-f654-11e9-802a-5aa538984bd8', 'MOCK_COGNITO_ID', 'NAME', 5, '%5B%7B%22MOCK%22%3A%22FORM%22%7D%5D', 'DESCRIPTION', 
          'REDIRECT', 'https://designshack.net/wp-content/uploads/placeholder-image.png', 'INSTRUCTIONS' )"
`;

exports[`createSurvey should add the survey to the database 2`] = `"INSERT INTO Tags(surveyId, tag) VALUES ( '575ea33e-f654-11e9-802a-5aa538984bd8', 'TAG1'),( '575ea33e-f654-11e9-802a-5aa538984bd8', 'TAG2');"`;

exports[`createSurvey should add the survey to the database 3`] = `
Object {
  "body": undefined,
  "headers": Object {
    "Access-Control-Allow-Credentials": true,
    "Access-Control-Allow-Origin": "*",
  },
  "statusCode": 200,
}
`;

exports[`createSurvey should respond with failure when an error occurs 1`] = `
Object {
  "body": undefined,
  "headers": Object {
    "Access-Control-Allow-Credentials": true,
    "Access-Control-Allow-Origin": "*",
  },
  "statusCode": 500,
}
`;
